### ISSUE TITLE
[Feature] use `http::Method` instead of re-implementing `Method`.

### ISSUE THREAD

**Issue by mendess** [cloudflare/workers-rs#426]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

This crate implements it's own [method type](https://docs.rs/worker/latest/worker/enum.Method.html) instead of using the more ecosystem standard `http::Method`. This makes it harder to integrate with `reqwest_wasm` and other crates that make use of the `http::Method` type.

Changing this now is a breaking change, which is annoying but I think it's worth it in the long run. Specially since `http` has just reached `1.0`! A non breaking alternative would be to add two from implementations in order to make the conversion easier.
```rs
impl From<http::Method> for Method {}
impl From<Method> for http::Method {}
```

What do you think?

**Comment by fwqaaq** [comment]
Yeah, and it can not create custom `Method`, which makes it completely impractical when I want to build webdav.

Maybe someone knows?

**Comment by kflansburg** [comment]
Fixed in #477
